### PR TITLE
Add WebUI port to readme-vars

### LINKED ISSUE #11: Ports are not listed in all applicable README sections
[linuxserverurl]: https://linuxserver.io
[linuxserver.io][linuxserverurl]

------------------------------

## Expected Behavior
The README should list required port configuration in parameters and examples sections. For example, see the https://github.com/linuxserver/docker-lidarr README.

## Current Behavior
The port for the web UI is only listed under application setup section.

### PR DESCRIPTION
**Pull request by rocketeerbkw**
[linuxserverurl]: https://linuxserver.io
[linuxserver.io][linuxserverurl]

------------------------------

 - [x] I have read the [contributing](https://github.com/linuxserver/docker-homeassistant/blob/main/.github/CONTRIBUTING.md) guideline and understand that I have made the correct modifications

------------------------------

## Description:
Adds port information to `readme-var.yml` so that the README has more complete information.

## Benefits of this PR and context:
While setting this up I followed the examples but couldn't connect to the web ui because the port wasn't included, I had to go back and add it in manually after the fact. This would make it work on the first go around.

## How Has This Been Tested?
I ran the [jenkins builder](https://github.com/linuxserver/docker-jenkins-builder#running-against-local-project) locally and verified the README.md is correct

## Source / References:
closes #11

### CODE REVIEW

**Comment by rocketeerbkw** [comment]
I looked at the reference vars file and noticed the "optional ports" is another way to do it and might be the correct way? It looks like the [wireshark container](https://github.com/linuxserver/docker-wireshark#parameters) uses this and it might be a better example than [lidarr](https://github.com/linuxserver/docker-lidarr#parameters). Under normal usage the `--net=host` should not require a port mapping, and the wireshark optional port also explicitly mentions that.

**Review by tobbenb — CHANGES_REQUESTED** [review]
It's not required, so please set it as optional with a good explanation so people understand they don't have to map the port if using host network.

**Comment by drizuid** [comment]
I think this needs a note that not using host will break mdns.

**Comment by tobbenb** [comment]
That is a good point driz

**Comment by rocketeerbkw** [comment]
So I should move everything to the `optional_parameters` section like the plex example (and mention mDNS) or keep the `opt_param_ports` and just add a note about mDNS?

**Comment by aptalca** [comment]
I think it's better to do opt param ports with notes about being required in bridge networking

**Comment by drizuid** [comment]
I agree, i think the big thing here is I dont think many use mdns in plex so there's no big loss, but i would venture to guess 75% or more use mdns in HA as most integrations rely on discovery.

**Comment by rocketeerbkw** [comment]
I moved and expanded the note about discovery into it's own section, based on the https://github.com/linuxserver/docker-tvheadend README

Editing_docker-homeassistant_README_md_at_main_·_rocketeerbkw_docker-homeassistant

File: readme-vars.yml [hunk 1]
```diff
@@ -25,11 +40,16 @@ param_env_vars:
...
-param_net_desc: "Shares host networking with container. Required for some devices to be discovered by Home Assistant."
+param_net_desc: "Shares host networking with container."
 param_device_map: true
 param_devices:
   - { device_path: "/path/to/device", device_host_path: "/path/to/device", desc: "For passing through USB, serial or gpio devices." }
 
+# optional container parameters
+opt_param_usage_include_ports: true
+opt_param_ports:
+  - { external_port: "8123", internal_port: "8123", port_desc: "Application WebUI, only use this if you are not using host mode." }
```
Comment (drizuid) [inline readme-vars.yml:51]: Please change the text to something beginning with optional, optionally, #optional or something similar so it looks like 
"Optionally assign the application webUI port, when not using host mode" or something. you'll notice our other optional fields look similar to what im saying.

File: readme-vars.yml [hunk 2]
```diff
@@ -25,11 +40,16 @@ param_env_vars:
 param_usage_include_ports: false
 param_usage_include_net: true
 param_net: "host"
-param_net_desc: "Shares host networking with container. Required for some devices to be discovered by Home Assistant."
+param_net_desc: "Shares host networking with container."
```
Comment (drizuid) [inline readme-vars.yml:43]: I see no need for this change, the original text is accurate.

Comment (rocketeerbkw) [reply readme-vars.yml:43]: You requested that this info be in a breakout section https://github.com/linuxserver/docker-homeassistant/pull/12#issuecomment-868858261

File: readme-vars.yml (hunk 1 above)
Comment (rocketeerbkw) [reply readme-vars.yml:51]: >you'll notice our other optional fields look similar to what im saying

But you don't conform to that wording in 89% of the [other READMEs](https://github.com/search?l=&p=1&q=org%3Alinuxserver+opt_param_ports+filename%3Areadme-vars.yml&ref=advsearch&type=Code)

Comment (Roxedus) [reply readme-vars.yml:51]: This is fine, our templating marks it as optional

File: readme-vars.yml [hunk 3]
```diff
@@ -14,6 +14,21 @@ available_architectures:
...
+    #### Host vs. Bridge
+
+    Home Assistant can [discover][hb0] and automatically configure
+    [zeroconf][hb1]/[mDNS][hb2] and [UPnP][hb3] devices on your network. In
+    order for this to work you must create the container with `--net=host`.
+
+    [hb0]: https://www.home-assistant.io/integrations/discovery/#mdns-and-upnp
+    [hb1]: https://en.wikipedia.org/wiki/Zero-configuration_networking
+    [hb2]: https://en.wikipedia.org/wiki/Multicast_DNS
+    [hb3]: https://en.wikipedia.org/wiki/Universal_Plug_and_Play
```
Comment (Roxedus) [inline readme-vars.yml:30]: While Github parses this markdown as expected, does DockerHub?

File: readme-vars.yml (hunk 2 above)
Comment (drizuid) [reply readme-vars.yml:43]: I prefer to keep both for the lazy users that won't scroll down. Letting them know up front that not using host is a breaking change is pretty important for our support sanity.

File: readme-vars.yml (hunk 3 above)
Comment (rocketeerbkw) [reply readme-vars.yml:30]: It's standard markdown, but if you know how to test docker hub I can

File: readme-vars.yml (hunk 2 above)
Comment (rocketeerbkw) [reply readme-vars.yml]: added it back

File: readme-vars.yml (hunk 3 above)
Comment (rocketeerbkw) [reply readme-vars.yml:30]: I built the readme locally and manually added it to a private docker hub repo, the links in question all worked

**Review by drizuid — APPROVED** [review]
i'm good with it now

**Review by Roxedus — APPROVED** [review]

**Pull request merged**